seatunnel-common/src/main/java/io/github/interestinglab/waterdrop/common/config/Common.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.interestinglab.waterdrop.common.config;
+
+import java.net.URISyntaxException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+public class Common {
+
+    private static final List<String> ALLOWED_MODES = Arrays.asList("client", "cluster");
+
+    private static Optional<String> MODE = Optional.empty();
+
+    public static boolean isModeAllowed(String mode) {
+        return ALLOWED_MODES.stream().anyMatch(m -> mode.toLowerCase().equals(m));
+    }
+
+    /**
+     * Set mode. return false in case of failure
+     */
+    public static Boolean setDeployMode(String m) {
+        if (isModeAllowed(m)) {
+            MODE = Optional.of(m);
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public static Optional<String> getDeployMode() {
+        return MODE;
+    }
+
+    /**
+     * Root dir varies between different spark master and deploy mode,
+     * it also varies between relative and absolute path.
+     * When running waterdrop in --master local, you can put plugins related files in $project_dir/plugins,
+     * then these files will be automatically copied to $project_dir/seatunnel-core/target and token in effect if you start waterdrop in IDE tools such as IDEA.
+     * When running waterdrop in --master yarn or --master mesos, you can put plugins related files in plugins dir.
+     */
+    public static Path appRootDir() {
+        if (MODE.equals(Optional.of("client"))) {
+            try {
+                String path = Common.class.getProtectionDomain().getCodeSource().getLocation().toURI().getPath();
+                return Paths.get(path).getParent().getParent().getParent();
+            } catch (URISyntaxException e) {
+                throw new RuntimeException(e);
+            }
+        } else if (MODE.equals(Optional.of("cluster"))) {
+            return Paths.get("");
+        } else {
+            return null;
```

Comment:
how about throw: 
throw new IllegalStateException("MODE not support : " + MODE.orElse("null"));